### PR TITLE
275 bugfix l3out reference site bd removal

### LINKED ISSUE #275: L3Out reference is removed when parameters are added/removed on BD
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v1.7.4 on linux_amd64
+ provider registry.terraform.io/ciscodevnet/mso v1.1.0 

### MSO version

* V 4.3.1

### APIC version and APIC Platform for Site Level Resources

* V 6.0(5h) on-prem

### Affected Resource(s)

* mso_schema_site_bd

### Terraform Configuration Files
```hcl
terraform {
  required_providers {
    mso = {
      source = "CiscoDevNet/mso"
    }
  }
}

provider "mso" {
  username = "" # <MSO username>
  password = "" # <MSO pwd>
  url      = "" # <MSO URL>
  insecure = true
  platform = "nd"
}


data "mso_site" "test_site" {
  name = "TEST_SITE"
}


data "mso_tenant" "test_tenant" {
  name         = "test_tenant"
}


data "mso_schema" "test_schema" {
  name = "test_schema"
}

data "mso_schema_template_vrf" "test_vrf" {
  schema_id = data.mso_schema.test_schema.id
  template  = "test_template"
  name      = "test_vrf"
} 


resource "mso_schema_template_bd" "ndo_bug_bd" {
  schema_id                       = data.mso_schema.test_schema.id
  template_name                   = "test_template"
  name                            = "ndo_bug_bd"
  display_name                    = "ndo_bug_bd"
  vrf_name                        = "test_vrf"
  vrf_schema_id                   = data.mso_schema_template_vrf.test_vrf.schema_id
  vrf_template_name               = data.mso_schema_template_vrf.test_vrf.template
  layer2_unknown_unicast          = "proxy"
  intersite_bum_traffic           = false
  optimize_wan_bandwidth          = true
  layer2_stretch                  = false
  layer3_multicast                = false
  multi_destination_flooding      = "flood_in_encap"
  ipv6_unknown_multicast_flooding = "optimized_flooding"
  unknown_multicast_flooding      = "optimized_flooding"
}


resource "mso_schema_site_bd" "foo_schema_site_bd" {
  schema_id     = data.mso_schema.test_schema.id
  bd_name       = mso_schema_template_bd.ndo_bug_bd.name
  template_name = "test_template"
  site_id       = data.mso_site.test_site.id
  host_route    = true
  svi_mac       = "00:22:BD:F8:AA:BB"
}

resource "mso_schema_site_bd_l3out" "foo_schema_site_bd_l3out" {
  schema_id     = data.mso_schema.test_schema.id
  template_name = "test_template"
  site_id       = data.mso_site.test_site.id
  bd_name       = "ndo_bug_bd"
  l3out_name    = "TEST_L3OUT"
}
```

### Debug Output

Debug output available upon request

### Expected Behavior

When host_route is added or removed, it is expected that only that attribute will be modified in NDO

### Actual Behavior

When host_route is added or removed, L3Out reference is completely removed from NDO

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

host_route = true and L3Out associated to BD

2. `terraform apply`

### Important Factoids

host_route = false

### References

* #0000

### PR DESCRIPTION
**Pull request by akinross**
fixes #275

### CODE REVIEW

File: mso/resource_mso_schema_site_bd.go [hunk 1]
```diff
@@ -279,14 +280,31 @@ func resourceMSOSchemaSiteBdUpdate(d *schema.ResourceData, m interface{}) error
...
+
+	err := setPatchPayloadToContainer(payloadCon, "replace", fmt.Sprintf("/sites/%s-%s/bds/%s/bdRef", siteId, templateName, bdName), bdRefMap)
+	if err != nil {
+		return err
+	}
+
+	err = setPatchPayloadToContainer(payloadCon, "replace", fmt.Sprintf("/sites/%s-%s/bds/%s/hostBasedRouting", siteId, templateName, bdName), host)
+	if err != nil {
+		return err
+	}
```
Comment (anvitha-jain) [inline mso/resource_mso_schema_site_bd.go:294]: do we need a condition before this? will it error out if there is no host?

**Review by anvitha-jain — CHANGES_REQUESTED** [review]

File: mso/resource_mso_schema_site_bd.go (hunk 1 above)
Comment (akinross) [reply mso/resource_mso_schema_site_bd.go:294]: The variable is initialised here so set to false so will always have a host_route: https://github.com/CiscoDevNet/terraform-provider-mso/pull/276/files#diff-86bef212dcb5a9532691f6ae8c3f0365cff9505f20b6642a9dcc55836c7a6da5R263.

**Review by lhercot — CHANGES_REQUESTED** [review]

**Review by lhercot — APPROVED** [review]
LGTM

**Review by anvitha-jain — APPROVED** [review]
LGTM

**Review by samiib — APPROVED** [review]
LGTM

**Pull request merged**